src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -708,26 +733,28 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
                         {
                             bool emitPdbFile = Arguments.EmitPdb && emitOptions.DebugInformationFormat != Emit.DebugInformationFormat.Embedded;
 
-                            using (var peStreamProvider = new CompilerEmitStreamProvider(this, finalPeFilePath))
-                            using (var pdbStreamProviderOpt = emitPdbFile ? new CompilerEmitStreamProvider(this, finalPdbFilePath) : null)
+                            var peStreamProvider = new CompilerEmitStreamProvider(this, finalPeFilePath);
+                            var pdbStreamProviderOpt = emitPdbFile ? new CompilerEmitStreamProvider(this, finalPdbFilePath) : null;
+
+                            success = compilation.SerializeToPeStream(
+                                moduleBeingBuilt,
+                                peStreamProvider,
+                                pdbStreamProviderOpt,
+                                testSymWriterFactory: null,
+                                diagnostics: diagnosticBag,
+                                metadataOnly: emitOptions.EmitMetadataOnly,
+                                cancellationToken: cancellationToken);
+
+                            peStreamProvider.Close(diagnosticBag);
```

Comment:
SerializeToPeStream does not throw. WritePeToStream might and SerializeToPeStream handles all expected exceptions.